### ISSUE TITLE
Class probabilities for Inception V3 model

### ISSUE THREAD

**Issue by attilatoth4** [tensorflow/serving#751]
Hi All,

First of all, thanks for working on this awesome project!

I followed the tutorial at https://www.tensorflow.org/serving/serving_inception, with the suggested Inception V3 model inside. It installs and works like a charm, but when I use this Go code for connecting to it from the outside:

```
imageAsBytesList := tf_core_example.BytesList{[][]byte{imageBytes}}
	featureBytesList := tf_core_example.Feature_BytesList{&imageAsBytesList}
	features := tf_core_example.Features{
		map[string]*tf_core_example.Feature{ "image/encoded": {&featureBytesList}},
	}
	example := tf_core_example.Example{&features}
	examples := tf_serving_model.ExampleList{[]*tf_core_example.Example{&example}}
	inputExampleList := tf_serving_model.Input_ExampleList{&examples}

	request := tf_serving_model.ClassificationRequest{
		&tf_serving_model.ModelSpec{"inception", &version, "serving_default"},
		&tf_serving_model.Input{&inputExampleList},
	}

	response, err := client.Classify(context.TODO(), &request)
```

I got this response:

```
bin/tensorflow_serving_client --image-file ./1200px-Elephas_maximus_\(Bandipur\).jpg
&[{tusker 10.483619} {Indian elephant, Elephas maximus 6.947593} {African elephant, Loxodonta africana 6.7184505} {chain 3.103896} {padlock 1.8196384}]
```

My aim would be to get the score for a class as a percentage (as outputted by the last Softmax layer in the Inception network). I have also tried with "serving_default" and "predict_image" signatures, and also with the `Predict` interface before, but all of these somehow yield the same results on my side.

The picture in question was https://upload.wikimedia.org/wikipedia/commons/thumb/9/98/Elephas_maximus_%28Bandipur%29.jpg/1200px-Elephas_maximus_%28Bandipur%29.jpg, I feel that with this model it shall be more than 10.4% that an elephant is present. I tried a few other pictures, but hardly got over 10.0 in the results.

Also, in the protoc-generated files the following comments are present for this GRPC service response:

```
// A single class.
type Class struct {
	// Label or name of the class.
	Label string `protobuf:"bytes,1,opt,name=label" json:"label,omitempty"`
	// Score for this class (e.g., **the probability** the item belongs to this
	// class).
	Score float32 `protobuf:"fixed32,2,opt,name=score" json:"score,omitempty"`
}
```

So the response theoretically should be percentage. Am I doing anything wrong here? How can I get the percentages if these above are correct?

Thanks in advance!

**Comment by attilatoth4** [comment]
It turned out this can be solved by adding the Softmax activation to `inception_saved_model.py` and rebuild the whole Docker container (see tutorial link in the post above).
My changes were in the file:
After line 80:
```
    ...
    # Run inference.
    logits, _ = inception_model.inference(images, NUM_CLASSES + 1)

    # Transform output to topK result.
    values, indices = tf.nn.top_k(logits, NUM_TOP_CLASSES)
    # Add Softmax activation
    **normalized_values = tf.nn.softmax(values)**
    ...
```
and where `values` was used before `normalized_values` can be used instead (line 126):
```
    ...
    scores_output_tensor_info = tf.saved_model.utils.build_tensor_info(values)
    ...
```
After converting the model using the modified `inception_saved_model.py` the example classification above changes to:
```
bin/tensorflow_serving_service --image-file ./1200px-Elephas_maximus_\(Bandipur\).jpg
&[{tusker 0.9495872} {Indian elephant, Elephas maximus 0.027660402} {African elephant, Loxodonta africana 0.02199597} {chain 0.00059232773} {padlock 0.00016398945}]
```
Not sure whether this is the simplest approach, there might be some other configuration I was not aware of to achieve this.

**Comment by gautamvasudevan** [comment]
Looks like this has a solution :)
